### PR TITLE
[ACA-3416] Add Claim/Release actions on adf task form

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/ACA-3416
Task form does not have claim/release actions as part the form actions.

**What is the new behaviour?**

Now if the Task been shared among the candidate, task form will show claim/release actions based on the task status

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services/src/lib/form/form.component.ts [hunk 1]
```diff
@@ -23,6 +23,12 @@ import { EcmModelService, NodeService, WidgetVisibilityService,
 import { Observable, of, Subject } from 'rxjs';
 import { switchMap, takeUntil } from 'rxjs/operators';
 
+@Component({
```
Comment (eromano) [inline lib/process-services/src/lib/form/form.component.ts:26]: can you move it in a separate file?

File: lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    it('Should be able to call claim task service', () => {
+        const claimTaskSpy = spyOn(taskListService, 'claimTask').and.returnValue(of({}));
+        const button = fixture.nativeElement.querySelector('button');
+        button.click();
+        expect(claimTaskSpy).toHaveBeenCalledWith(fixture.componentInstance.taskId);
+    });
+
+    it('Should be able to catch success event on click of claim button', async() => {
+        spyOn(taskListService, 'claimTask').and.returnValue(of({}));
```
Comment (eromano) [inline lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts:67]: can you make the test less dense and respect the rule https://w3.alfresco.com/confluence/pages/viewpage.action?spaceKey=AP&title=Code+Style+E2E 05-14

File: lib/process-services/src/lib/task-list/components/task-form/task-form.component.ts [hunk 3]
```diff
@@ -117,6 +117,14 @@ export class TaskFormComponent implements OnInit {
   @Output()
   cancel = new EventEmitter<void>();
 
+  /** Emitted when the task is claimed. */
+  @Output()
+  claim: EventEmitter<any> = new EventEmitter<any>();
```
Comment (eromano) [inline lib/process-services/src/lib/task-list/components/task-form/task-form.component.ts:122]: not sure this logic should be in the form

Comment (mauriziovitale) [reply lib/process-services/src/lib/task-list/components/task-form/task-form.component.ts:122]: we have the same on the task form cloud. The name of the event should be changed. @eromano https://github.com/Alfresco/alfresco-ng2-components/blob/develop/lib/process-services-cloud/src/lib/task/task-form/components/task-form-cloud.component.ts#L78

File: lib/process-services/src/lib/form/form.component.ts (hunk 1 above)
Comment (mauriziovitale) [reply lib/process-services/src/lib/form/form.component.ts:26]: I agree as we have for the cloud.

File: lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+ */
+
+import { Component, Output, EventEmitter } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { setupTestBed } from '@alfresco/adf-core';
+import { of } from 'rxjs';
+import { TaskListService } from '../../services/tasklist.service';
+import { ProcessTestingModule } from '../../../testing/process.testing.module';
+
+describe('ClaimTaskDirective', () => {
```
Comment (mauriziovitale) [inline lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts:25]: @sivakumar414ram just notice that the cloud has the same name, can you please put the correct one for Cloud?

File: lib/process-services/src/lib/task-list/components/task-form/task-form.component.ts (hunk 3 above)
Comment (mauriziovitale) [reply lib/process-services/src/lib/task-list/components/task-form/task-form.component.ts:122]: Let's call them as the one in the cloud

File: lib/process-services/src/lib/form/form.component.ts (hunk 1 above)
Comment (sivakumar414ram) [reply lib/process-services/src/lib/form/form.component.ts]: Moved to separate file

File: lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts (hunk 4 above)
Comment (sivakumar414ram) [reply lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts:25]: updated

File: lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts (hunk 2 above)
Comment (sivakumar414ram) [reply lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts:69]: Updated

File: docs/process-services-cloud/directives/unclaim-task-cloud.directive.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+---
+Title: Unclaim Task Cloud Directive
+Added: v3.1.0
```
Comment (arditdomi) [inline docs/process-services-cloud/directives/unclaim-task-cloud.directive.md:3]: If this is a new directive, it has to be the next ADF version which is v3.9.0

File: docs/process-services-cloud/directives/unclaim-task-cloud.directive.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+## Class members
+
+### Properties
+
+| Name | Type | Default value | Description |
+| ---- | ---- | ------------- | ----------- |
+| appName | `string` | "" | (Required) The name of the application. |
+| taskId | `string` |  | (Required) The id of the task. |
```
Comment (arditdomi) [inline docs/process-services-cloud/directives/unclaim-task-cloud.directive.md:25]: Can you add a default value for the taskId?

File: docs/process-services/directives/claim-task.directive.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+<button adf-claim-task [taskId]="taskId" (success)="onTaskClaimed()">Claim</button>
+```
+
+## Class members
+
+### Properties
+
+| Name | Type | Default value | Description |
+| ---- | ---- | ------------- | ----------- |
+| taskId | `string` |  | (Required) The id of the task. |
```
Comment (arditdomi) [inline docs/process-services/directives/claim-task.directive.md:24]: same here

File: docs/process-services/directives/unclaim-task.directive.md [hunk 8]
```diff
@@ -1,18 +1,18 @@
 ---
 Title: Unclaim Task Directive
-Added: v3.1.0
+Added: v3.9.0
```
Comment (arditdomi) [inline docs/process-services/directives/unclaim-task.directive.md:3]: I think the added should remain 3.1.0 and the Last reviewed can be updated

File: lib/process-services/src/lib/form/form.component.spec.ts [hunk 9]
```diff
@@ -1013,3 +1014,77 @@ describe('FormComponent', () => {
...
+
+    afterEach(() => {
+        fixture.destroy();
+    });
+
+    it('should be able to inject custom outcomes and click on custom outcomes', () => {
+        const onCustomButtonOneSpy = spyOn(customComponent, 'onCustomButtonOneClick').and.callThrough();
+        const buttonOneBtn = debugElement.query(By.css('#adf-custom-outcome-1'));
+        const buttonTwoBtn = debugElement.query(By.css('#adf-custom-outcome-2'));
+        expect(buttonOneBtn).not.toBeNull();
```
Comment (arditdomi) [inline lib/process-services/src/lib/form/form.component.spec.ts:1082]: can we have a new line always before the expectations?

File: lib/process-services/src/lib/task-list/components/task-form/task-form.component.spec.ts [hunk 10]
```diff
@@ -495,4 +499,139 @@ describe('TaskFormComponent', () => {
...
+        it('should not display the release button to logged in user if task is claimed by other candidate member', async() => {
+            getTaskDetailsSpy.and.returnValue(of(claimedByGroupMemberMock));
+
+            component.taskId = 'mock-task-id';
+            fixture.detectChanges();
+            await fixture.whenStable();
+
+            const unclaimButton = fixture.debugElement.query(By.css('[data-automation-id="adf-task-form-unclaim-button"]'));
+
+            expect(component.isTaskClaimedByCandidateMember()).toBeFalsy();
```
Comment (arditdomi) [inline lib/process-services/src/lib/task-list/components/task-form/task-form.component.spec.ts:568]: Can we use toBe(true) and toBe(false) in all the occurrences? https://vincenttunru.com/toBeTruthy-vs-toBe-true/

File: docs/process-services-cloud/directives/unclaim-task-cloud.directive.md (hunk 6 above)
Comment (sivakumar414ram) [reply docs/process-services-cloud/directives/unclaim-task-cloud.directive.md:25]: Do we really need to add default here?

File: lib/process-services/src/lib/task-list/components/task-form/task-form.component.spec.ts (hunk 10 above)
Comment (sivakumar414ram) [reply lib/process-services/src/lib/task-list/components/task-form/task-form.component.spec.ts:568]: Updated

File: docs/process-services-cloud/directives/unclaim-task-cloud.directive.md (hunk 5 above)
Comment (sivakumar414ram) [reply docs/process-services-cloud/directives/unclaim-task-cloud.directive.md]: It was old directive, i just changed the name. I am not sure should we keep old or new version @mauriziovitale What do you think?

Comment (mauriziovitale) [reply docs/process-services-cloud/directives/unclaim-task-cloud.directive.md]: it is fine

File: docs/process-services-cloud/directives/claim-task-cloud.directive.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+## Class members
+
+### Properties
+
+| Name | Type | Default value | Description |
+| ---- | ---- | ------------- | ----------- |
+| appName | `string` | " " | (Required) The name of the application. |
+| taskId | `string` | " " | (Required) The id of the task. |
```
Comment (arditdomi) [inline docs/process-services-cloud/directives/claim-task-cloud.directive.md:25]: This has to be double quotes without space. Like this `""`. Please update all the occurrences

**Pull request merged**